### ISSUE TITLE
Changing my address in metamask did not change my account in Giveth

### ISSUE THREAD

**Issue by laurenluz** [Giveth/giveth-2#893]
I had been previously signed into giveth with one address from metamask and an account I called "giveth love". I had since changed my metamask to my personal account - "Lauren B" in Giveth.

But Giveth still had me "signed in" with Giveth Love.

So I ended up donating from my personal account, rather than my "Giveth Love" account (in metamask, but on Giveth it shows as donated from Giveth Love because that is the account I was signed in with previously.

@mateodaza lmk if that makes sense.

**Comment by mateodaza** [comment]
Login will preserve after you change wallets in metamask as you are asked to make a signature - I don't believe this is a bug, if you want the future to recognize change and make a new signature we can do it. But I don't believe is the best UX

**Comment by laurenluz** [comment]
Login was preserved even after I closed giveth, changed wallets and then re-opened giveth in a new tab... and this is what made it confusing. Because I saw my login name on Giveth as the one associated with the account I wanted to donate from, and then when I was donating it was easy to overlook the fact that the account was not the donation account.

I think in TRACE we require a new signature when the user changes the wallet, you really don't think it would be good to include that here?

**Comment by mateodaza** [comment]
While we build this feature we decided that if you are logged in you'll be recognized as a donor even if you donate from a different source than your selected provider. For example you are logged in with metamask, you can give with portis or whatever but you'll still be recognized as the donor.

Changing this flow to require a signature every time, could make the experience tedious and actually slow down the donation process, but that's my opinion. We have to decide what brings more value, either allowing speed on multiple sessions or making it easier to donate from multiple places seen as one. 

It's all about UX - Personally I believe that if we let know the user he/she is making a donation "logged in as xxxx" would improve it a lot, and maybe bring back the "anonymous" option so it's more clear that it's gonna be public.

**Comment by laurenluz** [comment]
yes! what you're saying makes total sense, adding the signature sounds like it would make things clucky.

Why don't we add like a confirmation popup that says "you are making a donation of x amount from y wallet under z name to the project q"

we can even add something about givebacks to that popup too, when it's ready... some that says "you are eligible to receive up to 75% (or r USD) in GIVbacks"

if we like this idea I can come up with better wording.
